File: spring-data-microservice-transaction-sample/customer-service/src/main/java/sample/customer/domain/model/Customer.java
package sample.customer.domain.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Table(schema = "customer_service", value = "customers")
public class Customer {

  @Id
  public final int customerId;
  public final String name;
  public final int creditLimit;
  public final int creditTotal;

  public Customer(int customerId, String name, int creditLimit, int creditTotal) {
    this.customerId = customerId;
    this.name = name;
    this.creditLimit = creditLimit;
    this.creditTotal = creditTotal;
  }

  public Customer withCreditTotal(int newCreditTotal) {
    return new Customer(this.customerId, this.name, this.creditLimit, newCreditTotal);
  }
}

File: scalardb-dotnet-samples/microservice-transactions-sample-with-shared-cluster-with-linq/Common/CustomerService/Customer.cs
using System.ComponentModel.DataAnnotations.Schema;
using ScalarDB.Client.DataAnnotations;

namespace MicroserviceTransactionsSample.Common.CustomerService;

[Table("customer_service.customers")]
public class Customer
{
    [PartitionKey]
    [Column("customer_id", Order = 0)]
    public int Id { get; set; }

    [Column("name", Order = 1)]
    public string Name { get; set; } = "";

    [Column("credit_limit", Order = 2)]
    public int CreditLimit { get; set; }

    [Column("credit_total", Order = 3)]
    public int CreditTotal { get; set; }
}
